src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -294,6 +294,8 @@ CHIP_ERROR CopyTxtRecord(TxtFieldKey key, char * buffer, size_t bufferLen, const
     case TxtFieldKey::kCommissionerPasscode:
         return CopyTextRecordValue(buffer, bufferLen,
                                    static_cast<uint16_t>(params.GetCommissionerPasscodeSupported().value_or(false) ? 1 : 0));
+    case TxtFieldKey::kJointFabricMode:
```

Comment:
Yeah, but where is the actual ADD_TXT_RECORD bit. that would reach this code?  Hard to tell whether the `value_or` here makes sense without understanding that.

Which raises the question: was this tested with platform mdns?

Reply:
The original plan was to address [#38203](https://github.com/project-chip/connectedhomeip/issues/38203) in a separate PR. However, to simplify reviewing and testing, the changes have been included in this PR as part of the latest commit.